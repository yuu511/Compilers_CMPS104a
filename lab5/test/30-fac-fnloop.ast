ROOT "<<ROOT>>" 0.0.0 {0} 
FUNCTION "" 6.22.1 {0} void function 
TYPE_ID "" 6.22.1 {0} 
VOID "void" 6.22.1 {0} 
IDENT "fail" 6.22.6 {0} 
PARAM "(" 6.22.11 {0} 
TYPE_ID "" 6.22.12 {1} string variable param lval 
STRING "string" 6.22.12 {0} 
IDENT "expr" 6.22.19 {0} 
TYPE_ID "" 6.22.25 {1} string variable param lval 
STRING "string" 6.22.25 {0} 
IDENT "file" 6.22.32 {0} 
TYPE_ID "" 6.22.38 {1} int variable param lval 
INT "int" 6.22.38 {0} 
IDENT "line" 6.22.42 {0} 
FUNCTION "" 6.24.1 {1} void function 
TYPE_ID "" 6.24.1 {0} 
VOID "void" 6.24.1 {0} 
IDENT "putchr" 6.24.6 {0} 
PARAM "(" 6.24.13 {0} 
TYPE_ID "" 6.24.14 {2} int variable param lval 
INT "int" 6.24.14 {0} 
IDENT "chr" 6.24.18 {0} 
FUNCTION "" 6.25.1 {2} void function 
TYPE_ID "" 6.25.1 {0} 
VOID "void" 6.25.1 {0} 
IDENT "putint" 6.25.6 {0} 
PARAM "(" 6.25.13 {0} 
TYPE_ID "" 6.25.14 {3} int variable param lval 
INT "int" 6.25.14 {0} 
IDENT "num" 6.25.18 {0} 
FUNCTION "" 6.26.1 {3} void function 
TYPE_ID "" 6.26.1 {0} 
VOID "void" 6.26.1 {0} 
IDENT "putstr" 6.26.6 {0} 
PARAM "(" 6.26.13 {0} 
TYPE_ID "" 6.26.14 {4} string variable param lval 
STRING "string" 6.26.14 {0} 
IDENT "str" 6.26.21 {0} 
FUNCTION "" 6.28.1 {4} int function 
TYPE_ID "" 6.28.1 {0} 
INT "int" 6.28.1 {0} 
IDENT "getchr" 6.28.5 {0} 
PARAM "(" 6.28.11 {0} 
FUNCTION "" 6.29.1 {5} string function 
TYPE_ID "" 6.29.1 {0} 
STRING "string" 6.29.1 {0} 
IDENT "getstr" 6.29.8 {0} 
PARAM "(" 6.29.14 {0} 
FUNCTION "" 6.30.1 {6} string function 
TYPE_ID "" 6.30.1 {0} 
STRING "string" 6.30.1 {0} 
IDENT "getln" 6.30.8 {0} 
PARAM "(" 6.30.13 {0} 
FUNCTION "" 7.8.1 {7} int function 
TYPE_ID "" 7.8.1 {0} 
INT "int" 7.8.1 {0} 
IDENT "fac" 7.8.5 {0} 
PARAM "(" 7.8.9 {0} 
TYPE_ID "" 7.8.10 {8} int variable param lval 
INT "int" 7.8.10 {0} 
IDENT "n" 7.8.14 {0} 
BLOCK "{" 7.8.17 {8} 
TYPE_ID "" 7.9.4 {8} int variable local lval vreg (7.9.4)
INT "int" 7.9.4 {0} 
IDENT "f" 7.9.8 {0} 
INTCON "1" 7.9.12 {8} int variable local lval vreg 
WHILE "while" 7.10.4 {8} 
GT ">" 7.10.13 {8} int vreg 
IDENT "n" 7.10.11 {8} int variable param lval (7.8.10)
INTCON "1" 7.10.15 {8} int const 
BLOCK "{" 7.10.18 {8} 
'=' "=" 7.11.9 {8} int variable local lval vreg 
IDENT "f" 7.11.7 {8} int variable local lval (7.9.4)
'*' "*" 7.11.13 {8} int vreg 
IDENT "f" 7.11.11 {8} int variable local lval (7.9.4)
IDENT "n" 7.11.15 {8} int variable param lval (7.8.10)
'=' "=" 7.12.9 {8} int variable param lval vreg 
IDENT "n" 7.12.7 {8} int variable param lval (7.8.10)
'-' "-" 7.12.13 {8} int vreg 
IDENT "n" 7.12.11 {8} int variable param lval (7.8.10)
INTCON "1" 7.12.15 {8} int const 
RETURN "return" 7.14.4 {8} 
IDENT "f" 7.14.11 {8} int variable local lval (7.9.4)
FUNCTION "" 7.17.1 {8} int function 
TYPE_ID "" 7.17.1 {0} 
INT "int" 7.17.1 {0} 
IDENT "main" 7.17.5 {0} 
PARAM "(" 7.17.9 {0} 
BLOCK "{" 7.17.12 {9} 
TYPE_ID "" 7.18.4 {9} int variable local lval vreg (7.18.4)
INT "int" 7.18.4 {0} 
IDENT "n" 7.18.8 {0} 
INTCON "1" 7.18.12 {9} int variable local lval vreg 
WHILE "while" 7.19.4 {9} 
LE "<=" 7.19.13 {9} int vreg 
IDENT "n" 7.19.11 {9} int variable local lval (7.18.4)
INTCON "5" 7.19.16 {9} int const 
BLOCK "{" 7.19.19 {9} 
CALL "(" 7.20.14 {2} void function (6.25.1)
IDENT "putint" 7.20.7 {0} 
CALL "(" 7.20.19 {7} int function (7.8.1)
IDENT "fac" 7.20.15 {0} 
IDENT "n" 7.20.20 {9} int variable local lval (7.18.4)
CALL "(" 7.21.14 {1} void function (6.24.1)
IDENT "putchr" 7.21.7 {0} 
CHARCON "'\n'" 7.21.15 {9} int const 
'=' "=" 7.22.9 {9} int variable local lval vreg 
IDENT "n" 7.22.7 {9} int variable local lval (7.18.4)
'+' "+" 7.22.13 {9} int vreg 
IDENT "n" 7.22.11 {9} int variable local lval (7.18.4)
INTCON "1" 7.22.15 {9} int const 
RETURN "return" 7.24.4 {9} 
INTCON "0" 7.24.11 {9} int const 
