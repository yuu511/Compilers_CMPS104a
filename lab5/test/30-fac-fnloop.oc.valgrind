==24243== Memcheck, a memory error detector
==24243== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==24243== Using Valgrind-3.14.0 and LibVEX; rerun with -h for copyright info
==24243== Command: ../oc 30-fac-fnloop.oc
==24243== 
==24243== 
==24243== HEAP SUMMARY:
==24243==     in use at exit: 0 bytes in 0 blocks
==24243==   total heap usage: 579 allocs, 579 frees, 150,706 bytes allocated
==24243== 
==24243== All heap blocks were freed -- no leaks are possible
==24243== 
==24243== For counts of detected and suppressed errors, rerun with: -v
==24243== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
