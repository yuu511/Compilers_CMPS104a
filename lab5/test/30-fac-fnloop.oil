fac:       .function int 
           .param int n
           .local int f
           f = 1  
.wh0:      goto .od0 if n < 1
.do0:      $t0:i = f * n
           f = $t0:i  
           $t1:i = n - 1
           n = $t1:i  
           goto .wh0  
.od0
           return f
           return
           .end
main:      .function int 
           .local int n
           n = 1  
.wh1:      goto .od1 if n >= 5
.do1:      $t0:i = call fac(n)  
           call putint($t0:i)  
           call putchr('\n')  
           $t1:i = n + 1
           n = $t1:i  
           goto .wh1  
.od1
           return 0
           return
           .end
