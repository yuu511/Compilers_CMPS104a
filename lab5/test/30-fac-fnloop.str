string_set[   0]:    7296690267450279874 0x55dc584bcd58->"chr"
                   13348923631108115385 0x55dc584bd718->"putstr"
string_set[   5]:   12679825168387489523 0x55dc584bd278->"num"
string_set[   7]:    2297668033614959926 0x55dc584c0bf8->"0"
string_set[  10]:    1583576014053361350 0x55dc584bd8d8->"str"
                    6637313742931709005 0x55dc584bf218->"*"
                   13497578857207247412 0x55dc584bc318->"string"
                   11597697714117577063 0x55dc584beab8->"="
string_set[  12]:   13693284975010602715 0x55dc584be518->"fac"
string_set[  13]:   12868445110721718657 0x55dc584bc978->")"
string_set[  14]:    7995238595416485409 0x55dc584bffa8->"5"
string_set[  16]:   11911796782241914524 0x55dc584bb088->"void"
                   10159970873491820195 0x55dc584beb78->"1"
string_set[  19]:    5344662657899890615 0x55dc584bca38->";"
string_set[  20]:   15816506932648033945 0x55dc584bc258->"("
                   11868120863286193964 0x55dc584bf8f8->"main"
string_set[  25]:   11166701219533395393 0x55dc584bdef8->"getstr"
string_set[  29]:    1370730858159036685 0x55dc584c0a48->"+"
                   14635384904416252381 0x55dc584bf718->"return"
string_set[  34]:    7211130878751977441 0x55dc584bcb98->"putchr"
string_set[  37]:    6192799629375248351 0x55dc584c07c8->"'\n'"
                   12291392672316406109 0x55dc584bc898->"line"
string_set[  39]:   18413295556186398326 0x55dc584bd0b8->"putint"
string_set[  40]:   10319632053171904911 0x55dc584becf8->"while"
string_set[  41]:    9826907452181669771 0x55dc584be6d8->"n"
string_set[  43]:    7372855250431704371 0x55dc584bf518->"-"
string_set[  44]:      21606623943142546 0x55dc584be8b8->"{"
string_set[  45]:    6142509188972423790 0x55dc584bc498->""
string_set[  47]:   13686721134187162920 0x55dc584bf6b8->"}"
string_set[  57]:    9386170080689500517 0x55dc584bc3d8->"expr"
string_set[  58]:    4017852018407011111 0x55dc584bc658->"file"
                    3729804957429652673 0x55dc584b5d08->"<<ROOT>>"
string_set[  59]:   16579916199511856337 0x55dc584be1f8->"getln"
string_set[  63]:    9611428142580219387 0x55dc584bc158->"fail"
string_set[  64]:   12597418120160561823 0x55dc584bc7d8->"int"
string_set[  70]:    9201800968376702500 0x55dc584bdbf8->"getchr"
string_set[  71]:   14526079891305879776 0x55dc584bee38->">"
string_set[  75]:    8968846175329310707 0x55dc584be9f8->"f"
string_set[  76]:   15430489877420048373 0x55dc584bfee8->"<="
string_set[  78]:    4706206694200543916 0x55dc584bc198->","
load_factor = 0.506
bucket_count = 79
max_bucket_size = 4
