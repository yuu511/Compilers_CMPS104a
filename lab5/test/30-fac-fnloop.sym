main (7.17.1) {8} int function 
   n (7.18.4) {9} int variable local lval 0

getln (6.30.1) {6} string function 

fac (7.8.1) {7} int function 
   n (7.8.10) {8} int variable param lval 0
   f (7.9.4) {8} int variable local lval 1

putchr (6.24.1) {1} void function 

fail (6.22.1) {0} void function 

putint (6.25.1) {2} void function 

putstr (6.26.1) {3} void function 

getstr (6.29.1) {5} string function 

getchr (6.28.1) {4} int function 

