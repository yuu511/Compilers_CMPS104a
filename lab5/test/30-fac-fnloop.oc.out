CONDITIONAL WHILE STATEMENT SYMBOL TOK_GT 
CONDITIONAL WHILE STATEMENT SYMBOL TOK_LE 
